dotCMS/src/main/java/com/dotcms/integritycheckers/IntegrityUtil.java

```diff
@@ -243,79 +257,212 @@ public static void unzipFile(InputStream zipFile, String outputDir) throws Excep
+        addData.accept(integrityDataExecutionMetadata.getErrorMessage(), INTEGRITY_DATA_ERROR_MESSAGE);
+
+        try (FileOutputStream output = new FileOutputStream(statusFile)) {
+            statusData.store(output, null);
+        } catch (IOException e) {
+            throw new DotExecutionException(
+                    String.format("Could not save status to %s", statusFile.getAbsolutePath()),
+                    e);
+        }
+    }
+
+    /**
+     * Saves a integrity data generation metadata in a to-be-discovered location so it can be read by concurrent parts
+     * that need to know what is the status of the integrity data generation.
+     * Individual parameters are used to create a {@link IntegrityDataExecutionMetadata}.
+     *
+     * @param endpointId endpoint id
+     * @param requestId request id
+     * @param processStatus {@link IntegrityResource.ProcessStatus} instance to reflect the current status
+     * @param errorMessage error message associated to detected error
+     */
+    public static void saveIntegrityDataStatus(final String endpointId,
+                                               final String requestId,
+                                               final IntegrityResource.ProcessStatus processStatus,
+                                               final String errorMessage) {
+        saveIntegrityDataStatus(new IntegrityDataExecutionMetadata(endpointId, requestId, processStatus, errorMessage));
+    }
+
+    /**
+     * Saves a integrity data generation metadata in a to-be-discovered location so it can be read by concurrent parts
+     * that need to know what is the status of the integrity data generation.
+     * Individual parameters are used to create a {@link IntegrityDataExecutionMetadata}.
+     *
+     * @param endpointId endpoint id
+     * @param requestId request id
+     * @param processStatus {@link IntegrityResource.ProcessStatus} instance to reflect the current status
+     */
+    public static void saveIntegrityDataStatus(final String endpointId,
+                                               final String requestId,
+                                               final IntegrityResource.ProcessStatus processStatus) {
+        saveIntegrityDataStatus(endpointId, requestId, processStatus, null);
+    }
+
     /**
      * Creates all the CSV from End Point database table and store them inside
      * zip file.
      *
      * @param endpointId
      * @throws Exception
      */
-    public void generateDataToCheckZip(String endpointId) throws Exception {
-        File zipFile = null;
-
-        try {
-            if (!UtilMethods.isSet(endpointId))
-                return;
-
-            final String outputPath = ConfigUtils.getIntegrityPath() + File.separator + endpointId;
+    public static void generateDataToCheckZip(final String endpointId) throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)